carbon-design-system/carbon-addons-iot-react
# feat(datetimepickerv2): support both 12 and 24 hours time format

Closes #3595 

**Summary**

- indicate 12 hours or 24 hours by the definition of prop `dateTimeMask`, specifically if the hour mask is `HH`, it means 24 hours, and `hh` means 12 hours , despite of whether  `dateTimeMask` has AM/PM for not. 
- make sure both 12 hours and 24 hours are supported for single select and range select
- updated stories and tests

For example,

If `dateTimeMask` is `YYYY-MM-DD HH:mm` this is 24 hours. The field shows value like `2020-04-01 13:34`. `TimePickerDropDown` is 24hours format like `13:34`.

If `dateTimeMask` is `YYYY-MM-DD HH:mm A` this is 24 hours with AM/PM. The field shows value like `2020-04-01 13:34 PM`. `TimePickerDropDown` is 24hours format like `13:34`.

If `dateTimeMask` is `YYYY-MM-DD hh:mm A` this is 12 hours with AM/PM.  The field shows value like `2020-04-01 01:34 PM `. `TimePickerDropDown` is 24hours format like `01:34 PM`.


**Acceptance Test (how to verify the PR)**

For single select
- go to single select [story](https://deploy-preview-3596--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-datetimepickerv2--single-select)
- make sure is24hours is checked
- click on menu button to open flyout menu
- the time format on time picker should be 24 hours
- click on clock button to open time spinner
- the spinner should only show hours and minutes

For range select
- go to range select [story](https://deploy-preview-3596--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-datetimepickerv2--selected-absolute-with-new-time-spinner)
- make sure is24hours is checked
- click on menu button to open flyout menu
- the time format on time picker should be 24 hours
- click on clock button to open time spinner
- the spinner should only show hours and minutes





**Things to look for during review**
